src/model.py

```diff
@@ -50,6 +57,64 @@ def forward(self, inputs: torch.Tensor) -> torch.Tensor:
         return self.out(x)
 
 
+class ResidualBottleNeck(nn.Module):
+    '''
+    Class representing a convolutional residual block with a bottleneck
+    This class was built with reference to: 
+    https://github.com/kuangliu/pytorch-cifar/blob/master/models/resnet.py
+    '''
+
+    def __init__(self, num_channels: int, use_stem: bool = False, strides: int=1, factor: int=4):
+        '''
+        Creates a new instance of a Residual BottleNeck Block
+        @param: num_channels (int) - the number of output channels for all convolutions in the block
+        @param: use_stem (bool) - whether a 1x1 convolution is needed to downsample the residual
+        @param: strides (int) - the number of strides to use in the convolutions, defaults to 1
+        @param: factor (int) - the factor by which the input channels will be reduced for the bottleneck
+        '''
+        super().__init__()
+        self.num_channels = num_channels
+        self.use_stem = use_stem
+        self.strides = strides
+        self.factor = factor
+
+        # First convolutional layer with normalization
+        self.conv1 = nn.LazyConv2d(
+            num_channels//factor, kernel_size=1, padding=0)
```

Comment:
I think the decision to downsample by a factor is a great idea. That way we can specify different downsample factors if we want.